drivers/sensor/ite_vcmp_it8xxx2/vcmp_ite_it8xxx2.c

```diff
@@ -0,0 +1,398 @@
+/*
+ * Copyright (c) 2022 ITE Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_vcmp
+
+#include <device.h>
+#include <devicetree/io-channels.h>
+#include <drivers/adc.h>
+#include <drivers/sensor.h>
+#include <drivers/sensor/it8xxx2_vcmp.h>
+#include <dt-bindings/dt-util.h>
+#include <dt-bindings/sensor/it8xxx2_vcmp.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(vcmp_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+#define VCMP_REG_MASK		0x7
+#define VCMP_RESOLUTION		BIT(10)
+#define VCMP_MAX_MVOLT		3000
+
+/* Device config */
+struct vcmp_it8xxx2_config {
+	/* Voltage comparator x control register */
+	uintptr_t reg_vcmpxctl;
+	/* Voltage comparator x channel select MSB register */
+	uintptr_t reg_vcmpxcselm;
+	/* Voltage comparator scan period register */
+	uintptr_t reg_vcmpscp;
+	/* Voltage comparator x threshold data buffer MSB register */
+	uintptr_t reg_vcmpxthrdatm;
+	/* Voltage comparator x threshold data buffer LSB register */
+	uintptr_t reg_vcmpxthrdatl;
+	/* Voltage comparator status register */
+	uintptr_t reg_vcmpsts;
+	/* Voltage comparator status 2 register */
+	uintptr_t reg_vcmpsts2;
```

Comment:
why don't you store register pointers instead? 
you wouldn't need all such casts later:
```
volatile uint8_t *reg_vcmpsts = (uint8_t *)config->reg_vcmpsts;
```

am I missing something?

Reply:
Done